### PR TITLE
Allow for github dependencies

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR allows for dependencies on github branches, a la git+https://github.com/sncosmo/sncosmo.git@master#egg=sncosmo.

### CODE REVIEW

**Review by guynir42 — APPROVED** [review]
So, in this case it is up to the user to verify the requirement is installed?

**Comment by mcoughlin** [comment]
@guynir42 Yes.

File: tools/pip_install_requirements.py [hunk 1]
```diff
@@ -40,7 +41,13 @@ def pip(req_files):
 
 try:
     with status("Verifying Python package dependencies"):
-        pkg_resources.working_set.resolve([Requirement.parse(r) for r in requirements])
+        parsed_requirements = []
+        for r in requirements:
+            try:
+                parsed_requirements.append(Requirement.parse(r))
+            except InvalidRequirement:
```
Comment (stefanv) [inline tools/pip_install_requirements.py:48]: The only thing you can't check is the version, but you can still check whether the package is installed.

Comment (mcoughlin) [reply tools/pip_install_requirements.py:48]: @stefanv Are you thinking of some kind of magic string manipulation to get the package name to pass to Requirement.parse?

Comment (stefanv) [reply tools/pip_install_requirements.py:48]: I believe that `#egg=` is followed by a valid requirement specifier.

Comment (mcoughlin) [reply tools/pip_install_requirements.py]: Ok I just split it after that.

**Pull request merged**